packages/notifier/src/notifier.js

```diff
@@ -152,78 +78,22 @@ export const makeNotifierKit = (...initialStateArr) => {
- * @param {ERef<Subscriber<T>>} subscriberP
+ * @param {ERef<Subscriber<T>>} subscriber
  * @returns {Notifier<T>}
  */
-export const makeNotifierFromSubscriber = subscriberP => {
-  /** @type {bigint} */
-  let latestInboundCount;
-  /** @type {UpdateCount & bigint} */
-  let latestUpdateCount = 0n;
-  /** @type {WeakMap<PublicationRecord<T>, Promise<UpdateRecord<T>>>} */
-  const outboundResults = new WeakMap();
-
-  /**
-   * @param {PublicationRecord<T>} record
-   * @returns {Promise<UpdateRecord<T>>}
-   */
-  const translateInboundPublicationRecord = record => {
-    // Leverage identity preservation of `record`.
-    const existingOutboundResult = outboundResults.get(record);
-    if (existingOutboundResult) {
-      return existingOutboundResult;
-    }
-
-    latestInboundCount = record.publishCount;
-    latestUpdateCount += 1n;
-    const resultP = E.when(record.head, ({ value, done }) => {
-      if (done) {
-        // Final results have undefined updateCount.
-        return harden({ value, updateCount: undefined });
-      }
-      return harden({ value, updateCount: latestUpdateCount });
-    });
-    outboundResults.set(record, resultP);
-    return resultP;
-  };
-
+export const makeNotifierFromSubscriber = subscriber => {
   /**
    * @template T
    * @type {BaseNotifier<T>}
    */
-  const baseNotifier = Far('baseNotifier', {
-    getUpdateSince(updateCount = -1n) {
-      assert(
-        updateCount <= latestUpdateCount,
-        'argument must be a previously-issued updateCount.',
-      );
-
-      if (updateCount < latestUpdateCount) {
-        // Return the most recent result possible without imposing an unnecessary delay.
-        return E(subscriberP)
-          .subscribeAfter()
-          .then(translateInboundPublicationRecord);
-      }
-
-      // Return a result that follows the last-returned result,
-      // skipping over intermediate results if latestInboundCount
-      // no longer corresponds with the latest result.
-      // Note that unlike notifiers and subscribers respectively returned by
-      // makeNotifierKit and makePublishKit, this result is not guaranteed
-      // to follow the result returned when a non-latest updateCount is provided
-      // (e.g., it is possible for `notifierFromSubscriber.getUpdateSince()` and
-      // `notifierFromSubscriber.getUpdateSince(latestUpdateCount)` to both
-      // settle to the same object `newLatest` where `newLatest.updateCount`
-      // is one greater than `latestUpdateCount`).
-      return E(subscriberP)
-        .subscribeAfter(latestInboundCount)
-        .then(translateInboundPublicationRecord);
-    },
+  const baseNotifier = Far('Notifier', {
+    getUpdateSince: (updateCount = undefined) =>
+      E(subscriber).getUpdateSince(updateCount),
   });
 
   /** @type {Notifier<T>} */
   const notifier = Far('notifier', {
-    ...makeAsyncIterableFromNotifier(baseNotifier),
+    ...subscribeLatest(baseNotifier),
     ...baseNotifier,
 
     /**
```

Comment:
`getSharableNotifierInternals` is already provided by `makeNotifier`.
```suggestion
    ...makeNotifier(baseNotifier),
    ...baseNotifier,
```